docker/gcp_pudl_etl.sh

```diff
@@ -75,16 +76,26 @@ function notify_slack() {
     send_slack_msg "$message"
 }
 
-# # Run ETL. Copy outputs to GCS and shutdown VM if ETL succeeds or fails
-# 2>&1 redirects stderr to stdout.
-run_pudl_etl 2>&1 | tee $LOGFILE
+function run_deployment() {
+    # # Run ETL. Copy outputs to GCS and shutdown VM if ETL succeeds or fails
+    # 2>&1 redirects stderr to stdout.
+    run_pudl_etl 2>&1 | tee $LOGFILE
+
+    # Notify slack if the etl succeeded.
+    if [[ ${PIPESTATUS[0]} == 0 ]]; then
+        notify_slack "success"
+        copy_outputs_to_intake_bucket
+    else
+        notify_slack "failure"
+    fi
+}
 
-# Notify slack if the etl succeeded.
-if [[ ${PIPESTATUS[0]} == 0 ]]; then
-    notify_slack "success"
-    copy_outputs_to_intake_bucket
+if gsutil -q stat gs://pudl-etl-logs/$ACTION_SHA-$GITHUB_REF/pudl-etl.log; then
+    echo "Skipping deployment for $ACTION_SHA-$GITHUB_REF because outputs already exist for this commit.";
+    send_slack_msg "Skipping deployment for $ACTION_SHA-$GITHUB_REF because outputs already exist for this commit :cat:"
 else
-    notify_slack "failure"
+    echo "Running the ETL"
+    run_deployment
 fi
```

Comment:
This skipping logic could live in the github action but would be more complicated. See #1887

Reply:
I don't understand why we want to skip the deployment if the build failed. Don't we need to re-run the ETL from scratch if it previously failed?

Reply:
If the previous build failed and there aren't any new code changes, the next build will still fail.

Reply:
Unless the failures were due to external dependencies / APIs which seems sadly very common ☹️